python/pyspark/sql/functions/builtin.py

```diff
@@ -11241,13 +11241,18 @@ def sentences(
 ) -> Column:
     """
     Splits a string into arrays of sentences, where each sentence is an array of words.
-    The 'language' and 'country' arguments are optional, and if omitted, the default locale is used.
+    The `language` and `country` arguments are optional,
+    if they are both omitted, the default locale (the language is 'en' and the country is 'US')
+    is used, if the `country` is omitted, the default `country` ('') is used.
```

Comment:
and null means omitted?

Reply:
let's mention the default value for both parameters.

Reply:
It's surprising that we don't return null for this case

Reply:
I updated the comments like this:
```
 * The `lang` and `country` arguments are optional, their default values are all '',
 *  - When they are omitted:
 *    1. If they are both omitted, the `locale(language='', country='')` is used.
 *    2. If the `country` is omitted, the `locale(language, country='')` is used.
 *  - When they are null:
 *    1. If they are both `null`, the `Locale.US - locale(language='en', country='US')` is used.
 *    2. If the `language` is null and the `country` is not null,
 *       the `Locale.US - locale(language='en', country='US')` is used.
 *    3. If the `language` is not null and the `country` is null, the `locale(language)` is used.
 *    4. If neither is `null`, the `locale(language, country)` is used.
```

Reply:
> It's surprising that we don't return null for this case

Yes, the original expression `Sentences` did not inherit `NullIntolerant`. 
If we inherit it now, it may make this `breaking change`.

https://github.com/apache/spark/blob/9269a0bfed56429e999269dfdfd89aefcb1b7261/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L3347-L3351